apply/processor/gen.go

```diff
@@ -88,10 +91,36 @@ func (g *GenerateProcessor) GetPipeLine() ([]func(cluster *v2.Cluster) error, er
 		g.MountRootfs,
 		g.ApplyRegistry,
 		g.UnmountImage,
+		g.CmdAddRegistryHosts,
 	)
 	return todoList, nil
 }
 
+func (g *GenerateProcessor) CmdAddRegistryHosts(cluster *v2.Cluster) error {
+	regConfig := runtime.GetRegistryConfig(platform.DefaultMountClusterImageDir(cluster.Name), cluster.GetMaster0IP())
+	cmdAddSeaHubHosts := fmt.Sprintf(runtime.RemoteAddEtcHosts, regConfig.IP+" "+runtime.SeaHub, regConfig.IP+" "+runtime.SeaHub)
+	eg, _ := errgroup.WithContext(context.Background())
+	allList := cluster.GetAllIPList()
+	for _, ip := range allList {
+		ip := ip
+		eg.Go(func() error {
+			ssh, err := ssh.GetHostSSHClient(ip, cluster)
+			if err != nil {
+				return fmt.Errorf("new ssh client failed %v", err)
+			}
+			err = ssh.CmdAsync(ip, cmdAddSeaHubHosts)
+			if err != nil {
+				return fmt.Errorf("faild write sea.hub to /etc/hosts")
```

Comment:
fmt.Errorf("faild write sea.hub to /etc/hosts: %v", err)

When returned an detailed error, and you returned back an error and your wrapped an error. Please do not forget to take the detailed error in the returned err. Otherwise, no end-user would know what detailed error is, they have to seek for your support. At that time, the support team of sealer may not bear that. @Stevent-fei

Reply:
ok